ircd/m_gline.c

```diff
@@ -311,7 +311,7 @@ ms_gline(struct Client *cptr, struct Client *sptr, int parc, char *parv[])
 	if (!agline || /* gline creation, has to be the reason */
 	    /* trial-convert as lifetime, and if it doesn't fully convert,
 	     * it must be the reason */
-	    (!(lifetime = strtoul(parv[5], &tmp, 10)) && !*tmp)) {
+	    (((lifetime = strtoul(parv[5], &tmp, 10)) || 1) && *tmp != '\0')) {
```

Comment:
Why the `(... || 1)`?  And I'd rather have the right-hand side of && be either just `*tmp` or `(*tmp != '\0')` so that expressions are parenthesized similarly.

Reply:
Does this work?
`(lifetime = strtoul(parv[5], &tmp, 10), *tmp != '\0') {`
This may be one of the rare cases where the comma operator is useful :)

Reply:
Haha, indeed. Looks good to me. Issuing a new commit in a sec.